src/advanced_inv.cpp

```diff
@@ -1037,41 +1078,82 @@ void advanced_inventory::redraw_pane( side p )
     wrefresh( w );
 }
 
-bool advanced_inventory::move_all_items()
+// be explicit with the values
+enum aim_entry {
+    ENTRY_START     = 0,
+    ENTRY_VEHICLE   = 1,
+    ENTRY_MAP       = 2,
+    ENTRY_RESET     = 3
+};
+
+bool advanced_inventory::move_all_items(bool nested_call)
 {
     auto &spane = panes[src];
     auto &dpane = panes[dest];
 
     // AIM_ALL source area routine
     if(spane.get_area() == AIM_ALL) {
-        // future feature :-)
-        popup(_("You can't do that (yet!)"));
-        return false;
-        // make a copy of the current pane for below loop
+        // move all to `AIM_WORN' doesn't make sense (see `MAX_WORN_PER_TYPE')
+        if(dpane.get_area() == AIM_WORN) {
+            popup(_("You look at the items, then your clothes, and scratch your head..."));
+            return false;
+        }
+        // if the source pane (AIM_ALL) is empty, then show a message and leave
+        if(spane.items.empty()) {
+            popup(_("There are no items to be moved!"));
+            return false;
```

Comment:
This early return triggers a infinite loop:

- get some items on the floor around your character (I spawn some zombies and kill them, both via debug menu),
- open AIM, set active pane to "ALL", the other to some specific place,
- enter keys for "Move all items",
- items are moved,
- the "There are no items there" popup appears and stays.

Pressing enter or anything else does not seem to do anything (in fact, pressing enter or space closes the popup, but it is immediately recreated, so it appears to stay).

Once the last item is moved, `uistate.adv_inv_re_enter_move_all` is reset to `ENTRY_START`. When the AIM is (automatically) reopened, it thinks it's still processing (`is_processing` returns `true`) and calls `move_all_items` again.

At this point, the panes are as they were at the beginning, active pane set to "ALL", but it contains no items as all items are at the target place. `move_all_items` thinks there are no item to move, prints the popup and returns. But this does not affect `uistate.adv_inv_re_enter_move_all`, so `is_processing` still returns `true` and the whole thing starts again.

You should probably remove the early return and the popup. If there are no items, the action won't do anything (without message), the happens if there are no items and the "move item" command is entered.

Alternatively, it seems one can detect a user-initiated call to `move_all_items` by checking:
```C++
if( uistate.adv_inv_re_enter_move_all == ENTRY_START && uistate.adv_inv_aim_all_location == AIM_AROUND_BEGIN ) {
    // User just entered the move-all command
    popup();
    return;
}
// automatically called as part of the player activity.
// If the items list is empty, continue with the next place to move items from.
```